general/auth/2fa.rst

```diff
@@ -0,0 +1,144 @@
+compromised. With an additional factor, there are more hurdles an
+attacker needs to clear before they can access the account.
+
+Odoo currently supports :abbr:`2FA (two-factor authentication)` using
+*Time-based One-Time Passwords* (TOTP).
+
+This is the "something you own" category of factors: during the
+:abbr:`TOTP (time-based one-time password)` pairing process, you store
+a shared secret inside an *authenticator* (usually a program on your
+smartphone, although dedicated hardware devices are also
+available). At log-in, after you have input your password, Odoo will
+request a 6-digit *verification code*. This code is provided by your
+authenticator based on the shared secret and the current time.
+
+This means an attacker needs *both* to have guessed (or found) your
+password and to access (or steal) your authenticator, a more difficult
+proposition than either one or the other.
+
+Requirements
+============
+
+.. note:: these lists are just examples, they are not endorsements of
+          any specific software
+
+If you don't already have one, you will need to choose an
+authenticator.
+
+Phone-based authenticators are the easiest and most common so we will
+assume you'll pick and install one on your phone, examples include
+`authy <https://authy.com/>`_, `freeotp
+<https://freeotp.github.io/>`_, `google authenticator
+<https://support.google.com/accounts/answer/1066447?hl=en>`_,
+`lastpass authenticator <https://lastpass.com/auth/>`_, `microsoft
+authenticator
+<https://www.microsoft.com/en-gb/account/authenticator?cmp=h66ftb_42hbak>`_,
+...; password managers also commonly include :abbr:`2FA (two-factor
+authentication)` support e.g. `1password
+<https://support.1password.com/one-time-passwords/>`_, `bitwarden
+<https://bitwarden.com/help/article/authenticator-keys/>`_, ...
+
+For the sake of demonstration we will be using Google Authenticator
+(not because it is any good but because it is quite common).
+
+Setting up two-factor authentication
+====================================
+
+Once you have your authenticator of choice, go to the Odoo instance
+you want to setup :abbr:`2FA (two-factor authentication)`, the open
+:guilabel:`Preferences` (or :guilabel:`My Profile`):
+
+.. figure:: media/totp_setup/preferences.png
+    :align: center
+
+Open the :guilabel:`Account Security` tab, then click the
+:guilabel:`Enable two-factor authentication` button:
+
+.. figure:: media/totp_setup/sec_tab.png
+   :align: center
+
+Because this is a security-sensitive action, you will need to input
+your password:
+
+.. figure:: media/totp_setup/sec_enhanced.png
+   :align: center
+
+After which you will see this screen with a barcode:
+
+.. figure:: media/totp_setup/totp_scan.png
+   :align: center
+
+In most applications, you can simply *scan the barcode* via the
+authenticator of your choice, the authenticator will then take care of
+all the setup:
+
+.. figure:: media/totp_setup/scan_barcode.jpg
```

Comment:
This one is a moderately large JPG (100 KB). I'm not sure that it could be compressed further in PNG though.

Reply:
PNGs are almost always bigger than jpegs for pictures.